test/quantization/core/test_quantized_tensor.py

```diff
@@ -984,6 +1010,37 @@ def test_qtensor_fill_per_channel(self):
             self.assertEqual(q_filled.q_per_channel_scales(), scales)
             self.assertEqual(q_filled.q_per_channel_zero_points(), zero_points)
 
+    # adapted from test_qtensor_fill_per_channel and test_qtensor_fill_per_tensor_nhwc
+    def test_qtensor_fill_per_channel_nhwc(self):
+        dims = torch.randint(low=1, high=10, size=(4, )).tolist()
+        axis = 0
+        # adding a constant to avoid too small of a scale
+        scales = torch.rand(dims[axis], dtype=torch.float64) + 0.1
+        zero_points = torch.randint(low=0, high=10, size=(dims[axis], ))
+
+        ones = torch.ones(dims).to(torch.float)
+
+        qtypes = [torch.qint8, torch.quint8, torch.qint32]
+        vals2fill = [-1, 1, 2**32]  # positive, negative, overflow
+        memory_formats = [torch.contiguous_format, torch.channels_last]
+        devices = ["cpu", "cuda"] if TEST_CUDA else ["cpu"]
+        for qtype, val2fill, memory_format, device in itertools.product(qtypes, vals2fill, memory_formats, devices):
+            scales = scales.to(device)
+            zero_points = zero_points.to(device)
+            ones = ones.to(device)
+            q_filled = torch._empty_per_channel_affine_quantized(
+                dims, scales=scales, zero_points=zero_points, device=device,
+                axis=axis, dtype=qtype, memory_format=memory_format)
+            q_filled.fill_(val2fill)
+            # reference tensor for comparing q_filled
+            q_ref = torch.quantize_per_channel(ones * val2fill, scales=scales,
+                                               zero_points=zero_points, axis=axis, dtype=qtype)
+            self.assertEqual(q_filled.int_repr(), q_ref.int_repr())
+            self.assertEqual(q_filled.dequantize(), q_ref.dequantize())
+            # Make sure the scale and zero_point don't change
+            self.assertEqual(q_filled.q_per_channel_scales(), scales)
+            self.assertEqual(q_filled.q_per_channel_zero_points(), zero_points)
```

Comment:
have you tried using assertEqual directly?

Reply:
yeah I have, but it doesn't check for eqauivalence of qparams. e.g.,
```
        x = torch.zeros(4)
        scale1 = 1
        scale2 = 2
        zero_point = 0

        qx = torch.quantize_per_tensor(x, scale=scale1, zero_point=zero_point, dtype=torch.qint8)
        qy = torch.quantize_per_tensor(x, scale=scale2, zero_point=zero_point, dtype=torch.qint8)
        self.assertEqual(qx, qy)
        self.assertEqual(qx.dequantize(), qy.dequantize())
```

would pass even though the scales are different. i think we should have assertEqual check qparams as well